### ISSUE TITLE
 Add ability to clear all documents from a collection without dropping the collection and schema

### ISSUE THREAD

**Issue by shijithkjayan** [jaeyson/ex_typesense#28]
Hi,
I would like to request a feature to clear all documents from a collection without dropping the collection and its schema. This can be useful in scenarios where we want to reset the data in a collection but retain the schema for future use.

**Comment by shijithkjayan** [comment]
cc: @njausteve

**Comment by jaeyson** [comment]
Hi @shijithkjayan good idea. I'll circle back in couple of days and will let you know the progress. I'm replying here just to acknowledge the request. Thanks!

**Comment by jaeyson** [comment]
Hi @shijithkjayan. I'm not sure if you saw this thread: https://github.com/typesense/typesense/issues/1613

TLDR on that thread:

> > An alternative could be to simply delete the collection and recreate it

> This is indeed the workaround for now.

### workaround

[via docs](https://typesense.org/docs/26.0/api/documents.html#delete-by-query)

> To delete multiple documents by ID, you can use `filter_by=id: [id1, id2, id3]`.

> To delete all documents in a collection, you can use a filter that matches all documents in your collection. For eg, if you have an int32 field called popularity in your documents, you can use `filter_by=popularity:>0` to delete all documents. Or if you have a bool field called in_stock in your documents, you can use `filter_by=in_stock:[true,false]` to delete all documents.

I may have to update the `Document` module regarding this feature, since the docs gave us some idea how to match all documents in a collection leaving it up to developer's implementation.

What do you think?

**Comment by shijithkjayan** [comment]
Hi @jaeyson its indeed what I am doing right now. I drop a collection and then recreate it without any of the documents. I see that typesense itself doesn't support clearing an index and is suggesting a workaround. I am just wondering if you could add the logic for this in the library so those who use your library don't have to do it.

**Comment by jaeyson** [comment]
Hi @shijithkjayan kindly check this PR #33, specifically `test/document_test.exs`. Let me know your thoughts on this.
